documentation/STRUCTURE.md

```diff
@@ -0,0 +1,195 @@
+
+### API reference
+
+This is simply the generated documentation from our code. Currently we host this in readthedocs, but we want to include it on our webpage. A good example for this is [scikit-learn](https://scikit-learn.org/dev/modules/classes.html)
+
+### User guide
+
+The user guide explains all parts of Fairlearn by providing context that wouldn't fit into the code documentation such as mathematical derivations, but without using application-specific context (as we'd find it in the "case studies"). The guides are grouped by topic, e.g.
+
+1. What we mean by fairness in ML - should properly frame fairness as a sociotechnical challenge
+1. Assessment
+    1. Fairness definitions
+        1. ...
+    1. Metrics
+        1. ...
+    1. Dashboard
+        1. ...
+1. Mitigation
+    1. Postprocessing
+        1. Threshold Optimizer
+    1. Reductions methods
+        1. Exponentiated Gradient
+        1. Grid Search
+
+Importantly, the code samples should be minimal. For comprehensive examples we have the "Example Notebooks" section. In comparison, this section is more like a tutorial. It's about showing how to use our API while elaborating on mathemtical background that we can't explain in API documentation
+
+Examples:
+
+- [scikit-learn](https://scikit-learn.org/dev/user_guide.html)
+- [pandas](https://pandas.pydata.org/docs/user_guide/index.html)
+
+Of our current notebooks the following may be most suitable as "user guides":
+
+- [Group Metrics](https://github.com/fairlearn/fairlearn/blob/master/notebooks/Group%20Metrics.ipynb) - a great example for something that should be a user guide
+- [Grid Search for Binary Classification](https://github.com/fairlearn/fairlearn/blob/master/notebooks/Grid%20Search%20for%20Binary%20Classification.ipynb) - the purpose is mostly to show Grid Search's functionality; perhaps it may need to be trimmed down to the essentials about Grid Search
+- [Grid Search with Census Data](https://github.com/fairlearn/fairlearn/blob/master/notebooks/Grid%20Search%20with%20Census%20Data.ipynb) - similar to the previous notebook it covers Grid Search; we could leverage some of this for a user guide for Grid Search, or alternatively for the dashboard visualizations
+
+### Example Notebooks
+
+The purpose of the example notebooks is to walk through an application of Fairlearn in detail. Any application of a fairness toolkit needs to be done with great care while taking into account an entire range of concerns due to the sociotechnical nature of fairness. The showcased notebooks will provide the space to cover scenarios in depth. The focus is not only on showing example usage of the Fairlearn toolkit, but on how to approach fairness in ML in general. We may want to add a scenario even if it contains only few of Fairlearn's capabilities, but it otherwise demonstrates a great example of how to build AI responsibly.
```

Comment:
I think there's a tension between purposes here; some copy above say user guide notebooks say "*code samples should be minimal... it's about showing how to use our API.*"  Yet here it says examples should be "*taking into account an entire range of concerns due to the sociotechnical nature of fairness... the focus is not only on showing example usage of the Fairlearn toolkit, but on how to approach fairness in ML in general.*"

These strike me as key identity questions for the project :)

At the lowest level, it's simple to just do both and name the notebooks with different purposes.  But how the team chooses to answer this question, and what it chooses to *show* and not just *say* on the website, etc. will contribute to how people will perceive the project's brand, positioning, and research direction.  Clarifying that messaging could help with any goals around growing adoption and a community.

Reply:
User guides != example notebooks (if you have better names please suggest! I struggled with that initially as well.)

user guides just tell you how to call something, perhaps what the math background is, and what paper it comes from

example notebooks are full scenarios with (sometimes: real world) data where we can show that fairness is a tricky topic and needs to be treated with care, and that Fairlearn isn't a full solution.

Does that make more sense?

Reply:
Ah yeah I missed that, thanks! 👍 

When the time is right, I think it might be productive to make an example notebook where we can show "fairness is a tricky topic and needs to be treated with care, and *this is how fairlearn contributes to doing that more tricky work*."

Reply:
I'd go a step further than that and say that if we don't manage to make that clear in every notebook then we haven't done a good job.